core/templates/pages/story-editor-page/editor-tab/story-node-editor.directive.ts

```diff
@@ -132,10 +132,10 @@ angular.module('oppia').directive('storyNodeEditor', [
                 $scope.skillInfoHasLoaded = true;
                 $rootScope.$applyAsync();
               });
-            for (var idx in skillSummaries) {
+            for (var idx in skillSummaries) {  
               $scope.skillIdToSummaryMap[skillSummaries[idx].id] =
-                skillSummaries[idx].description;
-            }
+                skillSummaries[idx].description;              
+            }; 
```

Comment:
To get rid of the `for-in`, here are two options:
```
for (let summary of skillSummaries) { ... }
```

```
skillSummaries.forEach((summary) { ... })
```

Does that make sense?

Reply:
I have to replace summary with idx right??

Reply:
When we do `for-in` through an `Array`, the loop gives us an index `idx` starting from `0`. Now, when we do `for-of` through an `Array`, the loop gives us an item of the `Array`, which we can call a `summary`.